Sources/Misc/Purchases+async.swift

```diff
@@ -87,7 +87,7 @@ extension Purchases {
         return try await withCheckedThrowingContinuation { continuation in
             purchase(product: product,
                      promotionalOffer: promotionalOffer) { transaction, customerInfo, error, userCancelled in
-                continuation.resume(with: Result(customerInfo, error)
+                continuation.resume(with: Result(customerInfo, error?.ignoreIfPurchaseCancelled(userCancelled))
```

Comment:
feels like we're duplicating this code a lot, should we be moving this up the stack?

Reply:
We can't, we do want to send the error in the completion block based APIs (that's what broke PHC, that SK2 wasn't doing that), but we don't want to throw an error in the `async` APIs. Does that make sense?

Reply:
So to summarize:
- Completion block APIs need both the error and the cancellation parameter. This is consistent with SK1 and SK2, users can choose how to read cancellations
- `async` APIs can't have an error and a value, so throwing in case of cancellation seems like a bad API, especially because the returned value has a `userCancelled` boolean.

So we can't push this up because we only want to remove the error in `async` APIs. Does that make sense?